# this patch also speeds up the PGO run-time due to running tests in parallel
---
diff --git a/Makefile.pre.in b/Makefile.pre.in
index 35ca1a8..4782032 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -258,7 +258,12 @@ TCLTK_LIBS=	@TCLTK_LIBS@
 # The task to run while instrumented when building the profile-opt target.
 # We exclude unittests with -x that take a rediculious amount of time to
 # run in the instrumented training build or do not provide much value.
-PROFILE_TASK=-m test.regrtest --pgo
+# Solus: add -j4 to utilize 4 cpu threads to cut down on test run-time (0=max)
+# and add --ignore test_socket (since test_socket tends to randomly fail).
+# Note that each --ignore/-i is specified once for each ignore glob pattern.
+# Each glob is added to an ignorelist of strings in the test implementation.
+# Example: -i test_socket -i test_ssl -i test_a*
+PROFILE_TASK=-m test.regrtest --pgo -j4 --ignore test_socket
 
 # report files for gcov / lcov coverage report
 COVERAGE_INFO=	$(abs_builddir)/coverage.info
